cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -644,10 +644,15 @@ private SearchResults searchByCustomIndex(String namespaceId, Set<EntityTypeSimp
     // if the entity starts with _ then skip it unless the caller choose to showHidden.
     // This is done to hide entities from Tracker. See: CDAP-7910
 
-    NamespacedEntityId namespacedEntityId = EntityId.fromMetadataEntity(metadataEntity);
-    if (!showHidden && namespacedEntityId != null && namespacedEntityId.getEntityName().startsWith("_")) {
-      return Optional.absent();
+    try {
+      NamespacedEntityId namespacedEntityId = EntityId.fromMetadataEntity(metadataEntity);
```

Comment:
this can return null, right? You have to handle that.

Reply:
It will throw Illegal argument exception which I am catching and ignoring.